test/new_relic/http_client_test_cases.rb

```diff
@@ -531,27 +506,46 @@ def test_still_records_tt_node_when_request_fails
     # transaction in which the error occurs. That, coupled with the fact that
     # fixing it for old versions of Typhoeus would require large changes to
     # the instrumentation, makes us say 'meh'.
-    is_typhoeus = (client_name == 'Typhoeus')
-    if !is_typhoeus || (is_typhoeus && Typhoeus::VERSION >= '0.5.4')
-      evil_server = NewRelic::EvilServer.new
-      evil_server.start
+    skip 'Not tested with Typhoeus < 0.5.4' if typhoeus? && Typhoeus::VERSION < '0.5.4'
 
-      in_transaction do
-        begin
-          get_response("http://localhost:#{evil_server.port}")
-        rescue
-          # it's expected that this will raise for some HTTP libraries (e.g.
-          # Net::HTTP). we unfortunately don't know the exact exception class
-          # across all libraries
-        end
+    evil_server = NewRelic::EvilServer.new
+    evil_server.start
+
+    in_transaction do
+      begin
+        get_response("http://localhost:#{evil_server.port}")
+      rescue
+        # it's expected that this will raise for some HTTP libraries (e.g.
+        # Net::HTTP). we unfortunately don't know the exact exception class
+        # across all libraries
       end
+    end
 
-      last_node = find_last_transaction_node()
+    perform_last_node_assertions
 
-      assert_equal("External/localhost/#{client_name}/GET", last_node.metric_name)
+    evil_server.stop
+  end
 
-      evil_server.stop
-    end
+  def typhoeus?
+    client_name == 'Typhoeus'
+  end
+
+  def perform_last_node_assertions
+    last_node = find_last_transaction_node()
+
+    expected_name = typhoeus? ? 'Ethon' : client_name
```

Comment:
Should this still be there?

Reply:
Whoops! Lots of reversions needed for this file too. Thank you! Addressed with [914302d](https://github.com/newrelic/newrelic-ruby-agent/pull/2260/commits/914302d40142f9f396afa4d839b42c875f463da2)